prov/efa/src/efa_base_ep.h

```diff
@@ -65,6 +66,7 @@ struct efa_base_ep {
 	/* Only used by RDM ep type */
 	struct efa_qp *user_recv_qp; /* Separate qp to receive pkts posted by users */
 	struct efa_recv_wr *user_recv_wr_vec;
+	char err_msg[EFA_ERROR_MSG_BUFFER_LENGTH]; /* A large enough buffer to store CQ/EQ error data used by e.g. fi_cq_readerr */
```

Comment:
Paste my earlier comment here again

Do you have to use a persistent `err_msg` in `base_ep` ? If you read the `ofi_cq_write_err` code, you will find it will `mem_dup` the `err_data` you filled in `entry`. In other words, the `err_data` you passed can be safely released after `ofi_cq_write_err` returns. Just use a temporary char array in your code block that write the error cq.

Reply:
Do you prefer to remove err_msg from rdm_ep too?

Reply:
yes